### PR TITLE
[SQL Migration] Misc UI improvements

### PR DESCRIPTION
**Pull request by raymondtruong**
In this PR, we:
- add support for the `Trust server certificate` option
  - the user can specify this in ADS when connecting to their source instance, in the advanced properties:
   image
  - ADS allows the user to select `True`, `False`, or no selection at all -- the Start Migration request body now includes the `TrustServerCertificate` value, which defaults to `false`
  - sample request body:
   image
- automatically preselect all databases in the assessment results dialog
  - a common complaint was that users had to manually select all their desired databases twice, once at the beginning when they choose which databases to assess, then again when they choose which databases to actually migrate
- show warnings for in progress migrations in migration list
  - also update migration status dialog to show the correct status text
  - before / after: 
image
- fix a bug during performance data collection for SKU recommendation where changing and saving recommendation parameters while data collection was running would try to generate a recommendation (and error out) even though it wasn't ready yet
- fix a bug where a null reference exception in SKU recommendation would immediately appear when loading step 2 of the wizard, even if the user hadn't requested a recommendation yet
  - this was caused due to buggy save and close behavior which has now been updated
- fix an issue where the assessment results dialog was not scrollable, causing assessment issues with long names/descriptions to be cut off unless ADS is zoomed out
  - TO-DO: although the content is now scrollable, one of the black borders seems to be cut off when scrolled
- remove the info box informing users to ensure their backups were taken with checksum, as this requirement has now been lifted

### CODE REVIEW

File: extensions/sql-migration/src/api/utils.ts [hunk 1]
```diff
@@ -279,16 +279,15 @@ export function getMigrationStatusWithErrors(migration: azure.DatabaseMigration)
 	warningCount += properties.migrationFailureError?.message?.length > 0 ? 1 : 0;
 
 	// file upload blocking errors
-	warningCount += properties.migrationStatusDetails?.fileUploadBlockingErrors?.length ?? 0;
+	warningCount += properties.migrationStatusWarnings?.fileUploadBlockingErrorCount ?? 0;
```
Comment (brian-harris) [inline extensions/sql-migration/src/api/utils.ts:282]: when will is this property available?  is it deployed worldwide now?

Comment (brian-harris) [reply extensions/sql-migration/src/api/utils.ts:282]: for compatibility, will migrationStatusDetails still be returned for older migrations?

File: extensions/sql-migration/src/api/utils.ts [hunk 2]
```diff
@@ -279,16 +279,15 @@ export function getMigrationStatusWithErrors(migration: azure.DatabaseMigration)
...
 
 	// restore blocking reason
-	warningCount += properties.migrationStatusDetails?.restoreBlockingReason ? 1 : 0;
+	warningCount += properties.migrationStatusWarnings?.restoreBlockingReason ? 1 : 0;
 
-	// sql data copy errors
-	warningCount += properties.migrationStatusDetails?.sqlDataCopyErrors?.length ?? 0;
+	// complete restore error message
+	warningCount += properties.migrationStatusWarnings?.completeRestoreErrorMessage ? 1 : 0;
```
Comment (brian-harris) [inline extensions/sql-migration/src/api/utils.ts:288]: should this also check for string.length > 0 to assert it is a warning

File: extensions/sql-migration/src/models/stateMachine.ts [hunk 3]
```diff
@@ -889,7 +889,8 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 					dataSource: currentConnection?.serverName!,
 					authentication: this._authenticationType,
 					userName: this._sqlServerUsername,
-					password: this._sqlServerPassword
+					password: this._sqlServerPassword,
+					trustServerCertificate: currentConnection?.options.trustServerCertificate ?? false
```
Comment (brian-harris) [inline extensions/sql-migration/src/models/stateMachine.ts:893]: we should ensure we consider this in the other source connections we are making.  I added this parameter for validation as well:
const trustServerCertificate = currentConnection?.options['trustServerCertificate'] === true;

File: extensions/sql-migration/src/api/utils.ts (hunk 1 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/api/utils.ts:282]: yes, this was part of the 9/22 snap and should be deployed worldwide now. 

and yes, according to Austin's PR where he made the RP change, `MigrationStatusDetails` is still being populated as it was before (in one iteration it was removed, but then added back for backwards compatibility)

File: extensions/sql-migration/src/api/utils.ts (hunk 2 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/api/utils.ts:288]: this is what I meant to do, thanks for catching that

**Comment by brian-harris** [comment]
regarding the migration status column change,  what does the user see in the message box when they click on the status link.  Does the content need to be updated as well?
image

**Review by brian-harris — APPROVED** [review]

**Comment by raymondtruong** [comment]
> regarding the migration status column change, what does the user see in the message box when they click on the status link. Does the content need to be updated as well? 
image

It just reflects whatever's in the column (e.g. `"In progress (1 warning)"`), since the List Migration response may not contain the full list of warnings/errors, you'd have to go into the actual migration to view full details. So for now, just showing the migration state plus warning/error count is good enough

**Pull request merged**